modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -503,9 +550,21 @@ public Assignment getAssignment(User.ID userID, Application.Name applicationName
                 experimentBatch.getLabels().remove(experiment.getLabel());
             }
         }
+        if(LOGGER.isDebugEnabled()) LOGGER.debug("allAssignments: {} ", allAssignments);
```

Comment:
here is the code bits for logback 
```
private void filterAndLog_2(final String localFQCN,
      final Marker marker, final Level level, final String msg,
      final Object param1, final Object param2, final Throwable t) {

    final FilterReply decision = loggerContext.getTurboFilterChainDecision_2(
        marker, this, level, msg, param1, param2, t);

    if (decision == FilterReply.NEUTRAL) {
      if (effectiveLevelInt > level.levelInt) {
        return;
      }
    } else if (decision == FilterReply.DENY) {
      return;
    }

    buildLoggingEventAndAppend(localFQCN, marker, level, msg, new Object[] {
        param1, param2 }, t);
  }
```
as you can see before the log is logging anything it will check if the current level is necessary. if not necessary then nothing happens